// osu-api-extended/dist/types/enums.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Languages_enum = exports.Genres_enum = exports.Modes_enums = void 0;
var Modes_enums;
(function (Modes_enums) {
    Modes_enums[Modes_enums["osu"] = 0] = "osu";
    Modes_enums[Modes_enums["taiko"] = 1] = "taiko";
    Modes_enums[Modes_enums["fruits"] = 2] = "fruits";
    Modes_enums[Modes_enums["mania"] = 3] = "mania";
})(Modes_enums = exports.Modes_enums || (exports.Modes_enums = {}));
;
var Genres_enum;
(function (Genres_enum) {
    Genres_enum[Genres_enum["Unspecified"] = 0] = "Unspecified";
    Genres_enum[Genres_enum["Video Game"] = 1] = "Video Game";
    Genres_enum[Genres_enum["Anime"] = 2] = "Anime";
    Genres_enum[Genres_enum["Rock"] = 3] = "Rock";
    Genres_enum[Genres_enum["Pop"] = 4] = "Pop";
    Genres_enum[Genres_enum["Other"] = 5] = "Other";
    Genres_enum[Genres_enum["Novelty"] = 6] = "Novelty";
    Genres_enum[Genres_enum["Hip Hop"] = 7] = "Hip Hop";
    Genres_enum[Genres_enum["Electronic"] = 8] = "Electronic";
    Genres_enum[Genres_enum["Metal"] = 9] = "Metal";
    Genres_enum[Genres_enum["Classical"] = 10] = "Classical";
    Genres_enum[Genres_enum["Folk"] = 11] = "Folk";
    Genres_enum[Genres_enum["Jazz"] = 12] = "Jazz";
})(Genres_enum = exports.Genres_enum || (exports.Genres_enum = {}));
var Languages_enum;
(function (Languages_enum) {
    Languages_enum[Languages_enum["Any"] = 0] = "Any";
    Languages_enum[Languages_enum["English"] = 1] = "English";
    Languages_enum[Languages_enum["Chinese"] = 2] = "Chinese";
    Languages_enum[Languages_enum["French"] = 3] = "French";
    Languages_enum[Languages_enum["German"] = 4] = "German";
    Languages_enum[Languages_enum["Italian"] = 5] = "Italian";
    Languages_enum[Languages_enum["Japanese"] = 6] = "Japanese";
    Languages_enum[Languages_enum["Korean"] = 7] = "Korean";
    Languages_enum[Languages_enum["Spanish"] = 8] = "Spanish";
    Languages_enum[Languages_enum["Swedish"] = 9] = "Swedish";
    Languages_enum[Languages_enum["Russian"] = 10] = "Russian";
    Languages_enum[Languages_enum["Polish"] = 11] = "Polish";
    Languages_enum[Languages_enum["Instrumental"] = 12] = "Instrumental";
    Languages_enum[Languages_enum["Unspecified"] = 13] = "Unspecified";
    Languages_enum[Languages_enum["Other"] = 14] = "Other";
})(Languages_enum = exports.Languages_enum || (exports.Languages_enum = {}));
