// enka-network-api/dist/models/artifact/Artifact.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const ArtifactData_1 = __importDefault(require("./ArtifactData"));
const ArtifactSplitSubstat_1 = __importDefault(require("./ArtifactSplitSubstat"));
const StatProperty_1 = __importDefault(require("../StatProperty"));
const config_file_js_1 = require("config_file.js");
/**
 * @en Artifact
 */
class Artifact {
    /**
     * @param data
     * @param enka
     */
    constructor(data, enka) {
        var _a, _b, _c;
        this.enka = enka;
        this._data = data;
        const json = new config_file_js_1.JsonReader(this._data);
        const reliquary = json.get("reliquary");
        this.artifactData = ArtifactData_1.default.getById(json.getAsNumber("itemId"), enka);
        this.level = reliquary.getAsNumber("level");
        // const mainstatId = reliquary.getAsNumber("mainPropId");
        // const mainstatFightProp: FightProp = enka.cachedAssetsManager.getGenshinCacheData("ReliquaryMainPropExcelConfigData").findArray((_, m) => m.getAsNumber("id") === mainstatId)?.[1].getAsString("propType") as FightProp;
        const mainstatFightProp = json.getAsString("flat", "reliquaryMainstat", "mainPropId");
        const levelInfo = (_a = enka.cachedAssetsManager.getGenshinCacheData("ReliquaryLevelExcelConfigData").findArray((_, value) => value.getAsNumberWithDefault(0, "rank") === this.artifactData.stars && value.getAsNumber("level") === this.level)) === null || _a === void 0 ? void 0 : _a[1];
        const mainstatData = (_b = levelInfo.get("addProps").findArray((_, p) => p.getAsString("propType") === mainstatFightProp)) === null || _b === void 0 ? void 0 : _b[1];
        this.mainstat = new StatProperty_1.default(mainstatFightProp, mainstatData.getAsNumber("value"), enka);
        const splitSubStats = reliquary.has("appendPropIdList") ? (_c = reliquary.get("appendPropIdList")) === null || _c === void 0 ? void 0 : _c.mapArray((_, id) => ArtifactSplitSubstat_1.default.getById(id.getAsNumber(), enka)) : [];
        this.substats = {
            total: StatProperty_1.default.sumStatProperties(splitSubStats, enka),
            split: splitSubStats,
        };
    }
}
exports.default = Artifact;
